barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/rom_ram_logic.cpp

```diff
@@ -392,16 +434,15 @@ void RomRamLogic_<ExecutionTrace>::create_final_sorted_RAM_gate(CircuitBuilder*
 {
     record.record_witness = builder->add_variable(0);
     // Note: record the index into the block that contains the RAM/ROM gates
-    record.gate_index = builder->blocks.memory.size(); // no -1 since we havent added the gate yet
+    record.gate_index = builder->blocks.memory.size(); // no -1 since we _haven't_ added the gate yet
 
-    // TODO(https://github.com/AztecProtocol/barretenberg/issues/879): This method used to add a single arithmetic gate
-    // with two purposes: (1) to provide wire values to the previous RAM gate via shifts, and (2) to perform a
-    // consistency check on the value in wire 1. These two purposes have been split into a dummy gate and a simplified
-    // arithmetic gate, respectively. This allows both purposes to be served even after arithmetic gates are sorted out
-    // of sequence with the RAM gates.
+    // NOTE: This method used to add a single arithmetic gate with two purposes: (1) to provide wire values to the
```

Comment:
When we initially moved to the "sorted" execution trace it was unclear whether we'd eventually want to "undo" the addition of these unconstrained gates. Now its clear that its well worth the tradeoff. I think the language here can now just plainly state the puprose of unconstrained gate without referencing how things used to be